acvm-repo/acvm/src/compiler/optimizers/redundant_range.rs

```diff
@@ -157,20 +175,38 @@ impl<F: AcirField> RangeOptimizer<F> {
     }
 }
 
+fn memory_block_implied_max_bits(init: &[Witness]) -> u32 {
+    8 * size_of::<usize>() as u32 - init.len().leading_zeros()
+}
```

Comment:
```suggestion
fn memory_block_implied_max_bits(init: &[Witness]) -> u32 {
    32 - (init.len() as u32).leading_zeros()
}
```
Why bother with `size_of::<usize>()`? This will potentially change the calculation with a different target too. I find an explicit constant more clear

Reply:
I'm using this as I expect `init.len().leading_zeros()` to change to match any change in usize.

Reply:
I see how `init.len().leading_zeros()` will change to match a change in usize but couldn't this still then return something above 32 bits which is our usize in Noir. I don't foresee how this would be unsafe so I won't block on this, but just feels weird to rely on Rust's usize when it changes with the target while Noir's usize does not.